tezos/api/src/ocaml_conv/to_ocaml.rs

```diff
@@ -0,0 +1,297 @@
+        forking_block_hash: OCamlBlockHash,
+        test_chain_id: OCamlBytes,
+    ) -> ForkingTestchainData;
+
+    alloc fn alloc_begin_construction_request(
+        chain_id: OCamlBytes,
+        predecessor: BlockHeader,
+        protocol_data: Option<OCamlBytes>,
+    ) -> BeginConstructionRequest;
+
+    alloc fn alloc_validate_operation_request(
+        prevalidator: PrevalidatorWrapper,
+        operation: Operation,
+    ) -> ValidateOperationRequest;
+
+    alloc fn alloc_json_rpc_request(
+        body: OCamlBytes,
+        context_path: OCamlBytes,
+    ) -> JsonRpcRequest;
+
+    alloc fn alloc_protocol_json_rpc_request(
+        block_header: BlockHeader,
+        chain_id: OCamlBytes,
+        chain_arg: OCamlBytes,
+        request: JsonRpcRequest,
+        ffi_service: FfiRpcService,
+    ) -> ProtocolJsonRpcRequest;
+
+    // Headers
+
+    alloc fn alloc_block_header_shell_header(
+        level: OCamlInt32,
+        proto_level: OCamlInt,
+        predecessor: OCamlBlockHash,
+        timestamp: OCamlInt64,
+        validation_passes: OCamlInt,
+        operations_hash: OCamlOperationListListHash,
+        fitness: OCamlList<OCamlBytes>,
+        context: OCamlContextHash,
+    ) -> BlockHeaderShellHeader;
+
+    alloc fn alloc_block_header(
+        shell: BlockHeaderShellHeader,
+        protocol_data: OCamlBytes,
+    ) -> BlockHeader;
+
+    alloc fn alloc_operation_shell_header(
+        branch: OCamlBlockHash,
+    ) -> OperationShellHeader;
+
+    // Operation
+
+    alloc fn alloc_operation(
+        shell: OperationShellHeader,
+        proto: OCamlBytes,
+    ) -> Operation;
+
+    // Wrappers
+
+    alloc fn alloc_prevalidator_wrapper(
+        chain_id: OCamlBytes,
+        protocol: OCamlProtocolHash,
+    ) -> PrevalidatorWrapper;
+
+    // Hashes
+
+    alloc fn alloc_operation_list_list_hash(hash: OCamlBytes) -> OCamlOperationListListHash;
+    alloc fn alloc_operation_hash(hash: OCamlBytes) -> OCamlOperationHash;
+    alloc fn alloc_block_hash(hash: OCamlBytes) -> OCamlBlockHash;
+    alloc fn alloc_context_hash(hash: OCamlBytes) -> OCamlContextHash;
+    alloc fn alloc_protocol_hash(hash: OCamlBytes) -> OCamlProtocolHash;
+}
+
+unsafe impl ToOCaml<ApplyBlockRequest> for ApplyBlockRequest {
+    fn to_ocaml(&self, token: OCamlAllocToken) -> OCamlAllocResult<ApplyBlockRequest> {
+        let operations: Vec<Vec<FfiOperation>> = self
+            .operations
+            .iter()
+            .map(|ops| ops.iter().map(|op| FfiOperation(op.clone())).collect())
+            .collect();
+        unsafe {
+            alloc_apply_block_request(
+                token,
+                &self.chain_id,
+                &FfiBlockHeader(self.block_header.clone()),
+                &FfiBlockHeader(self.pred_header.clone()),
+                &self.max_operations_ttl,
+                &operations,
+            )
+        }
+    }
+}
+
+unsafe impl ToOCaml<ApplyBlockResponse> for ApplyBlockResponse {
```

Comment:
@tizoc 
Bruno, could we refactor this also with macro `to_from_ocaml_block_mapping` like we have `impl_from_ocaml_block_mapping`, than it will look like the same thing

Reply:
maybe after that, we could join `impl_from_ocaml_block_mapping` and `impl_to_ocaml_block_mapping` behind one macro like `impl_ocaml_block_mapping` where `from` and `to` conversion could be on one place,
and also maybe than we could generate inside `impl_ocaml_block_mapping` some test for conversion from/to automatically, so if we add new structs it will need some test_roundrobin stuff on ocaml side

but this is just my thought, lets just refactor it to `impl_to_ocaml_block_mapping` at first

Reply:
It would work for very simple cases like this one, where nothing has to be done with the fields, and there are no "FFI wrapper" structs involved, but for all others I just think it would make things less readable. The case above for example already needs to perform extra work, and is not a direct mapping of fields. I don't think it adds as much value here as it did for the implementation of the `FromOCaml` trait.

Regarding the tests, I thought about this a while yesterday, and I think I will just have to write all that manually, couldn't figure a good way to automate it, at least not with what we have right now. My plan is to write some functions on the OCaml side that takes one argument for each field of the structs, and construct an instance of that struct (in OCaml). In addition to those fields, as it's first argument this function would receive an instance of that struct, but constructed by Rust, and then it would compare these two structs structurally to verify that they are equal.

I will start working on that once everything else related to this PR is solved.